core/cleaner/src/main/java/com/cognifide/aet/cleaner/processors/GetMetadataArtifactsProcessor.java

```diff
@@ -104,19 +111,24 @@ public void process(Exchange exchange) throws Exception {
       stepArtifacts.add(stepResult.getArtifactId());
       stepArtifacts.addAll(traverseComparators(step));
     }
-    if (step.getPattern() != null) {
-      stepArtifacts.add(step.getPattern());
+    if (step.getPatterns() != null) {
+      Set<String> patterns = step.getPatterns().stream()
+          .filter(Objects::nonNull)
+          .map(Pattern::getPattern)
```

Comment:
```java
public Set<Pattern> getPatterns() {
    if (patterns == null) {
      patterns = new HashSet<>();
    }
```
Is it ` if (step.getPatterns() != null) ` necessary? In addition, you are filtering every null object ;)

Reply:
You are right, it's not necessary, thanks for commenting!